### DISCUSSION TITLE
post/2022-12-31-golang-dev-tools/

### DISCUSSION THREAD

**Discussion by giscus** [zcong1993/blog#76] [Announcements]
# post/2022-12-31-golang-dev-tools/

本文介绍用 Taskfile 来管理 golang 开发工具.

https://blog.cong.moe/post/2022-12-31-golang-dev-tools/

**Comment by PeterlitsZo** [comment]
从依赖多个工具变成了依赖 Taskfile 对吧 :cry:

JS 社区是真的不错，将所有依赖都放在了 `package.json` 中，相反，在 Go 中我们只能求助文档、第三方工具（Makefile，Taskfile）或者脚本。

标准化挺不错的，但是标准化也并没有那么简单就能实现，所以 Taskfile 很不错，但是想要在 Go 生态中推广成为事实标准感觉还是挺困难的。（捂脸）

**Reply by zcong1993** [reply]
所以一个东西官方不提供解决方案想统一就太难了。只能简单追求团队小范围的共识统一了。

**Comment by ttys3** [comment]
如果不需要模块化的功能，只是简单的一级 Makefile, justfile 可能更适合：

https://nguyenhuythanh.com/posts/taskfile-vs-justfile/

**Reply by zcong1993** [reply]
这里不光是利用 Taskfile 的任务依赖, 还想利用它的一些前置 check 来保证依赖工具安装, 不会重复安装并且在需要更新的时候自动更新.
